### PR TITLE
fix(agent): return non-200 status when > 0 failures in bulk op

### PR DESCRIPTION
**Pull request by ThatOneBro**

### CODE REVIEW

File: packages/server/src/fhir/operations/utils/agentutils.ts [hunk 1]
```diff
@@ -121,9 +123,13 @@ export async function handleBulkAgentOperation(
           agents[i]
         )
       );
+      failureCount += 1;
       continue;
     }
     const [outcome, params] = result.value;
+    if (!isOk(outcome)) {
+      failureCount += 1;
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/utils/agentutils.ts:131]: I recommend we save the `outcome` here and use it to determine status.

`++`?

File: packages/server/src/fhir/operations/agentbulkstatus.test.ts [hunk 2]
```diff
@@ -233,7 +233,7 @@ describe('Agent/$bulk-status', () => {
       .get('/fhir/R4/Agent/$bulk-status')
       .query({ name: 'Test Agent 2' })
       .set('Authorization', 'Bearer ' + accessToken);
-    expect(res.status).toBe(200);
+    expect(res.status).toBe(500);
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/agentbulkstatus.test.ts:236]: We should never intentionally return a 500 - that should be reserved for unexpected errors.

File: packages/server/src/fhir/operations/utils/agentutils.ts [hunk 3]
```diff
@@ -132,7 +138,7 @@ export async function handleBulkAgentOperation(
   }
 
   return [
-    allOk,
+    failureCount === 0 ? allOk : serverError(new Error(`${failureCount} sub-operations failed`)),
```
Comment (codyebberson) [inline packages/server/src/fhir/operations/utils/agentutils.ts:141]: Use the saved `outcome` here rather than `serverError`

**Review by codyebberson — CHANGES_REQUESTED** [review]

File: packages/server/src/fhir/operations/utils/agentutils.ts (hunk 1 above)
Comment (ThatOneBro) [reply packages/server/src/fhir/operations/utils/agentutils.ts:131]: It's possible that you get multiple different types of outcomes, should we just save the first one?

**Pull request closed without merging**